src/controllers/favoriteController.js

```diff
@@ -0,0 +1,116 @@
+const userModel = require("../models/userSchema");
+
+/******************************************************************************
+ * @addFavoriteQuiz
+ * @route /api/v1/favorites/add
+ * @method POST
+ * @description add favorite quiz from the user in the favorites array in database
+ * @body quizId
+ * @returns favorites array
+ *******************************************************************************/
+
+//Add quiz as favorite
+const addFavoriteQuiz = async (req, res) => {
+  const { userId } = req.user;
+  const { quizId } = req.body;
+
+  try {
+    const user = await userModel.findById(userId);
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    // Add quiz in the favorites array
+    user.favorites.push(quizId);
+
+    await user.save();
+
+    //update cookies
+    const token = user.jwtToken();
+    const cookiesOptions = {
+      maxAge: 24 * 60 * 60 * 1000,
+      httpOnly: true,
+    };
+    res.cookie("token", token, cookiesOptions);
+
+    res.status(200).json({
+      success: true,
+      message: "Quiz added from favorites successfully",
+    });
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+/******************************************************************************
+ * @getUserFavoriteQuiz
+ * @route /api/v1/favorites
+ * @method GET
+ * @description display favorites quizz from the user
+ * @body
+ * @returns favorites array
+ *******************************************************************************/
+
+const getUserFavorites = async (req, res) => {
+  const { userId } = req.user;
+
+  try {
+    const user = await userModel.findById(userId);
+
+    if (!user) {
+      return res.status(404).json({ error: "User not found" });
+    }
+
+    // Return favorite list to the user.
+    res.json({ favorites: user.favorites });
```

Comment:
I don't think this is an issue since the number of quizzes is small, but for future reference when making favorite endpoints, it's much more convenient for the front-end if you return the actual quizzes in the get favorites call. Right now the front-end will either have to 1. get the IDs from the get favorites call then make a request to the backend to get the quiz details for each favorite or 2. get the IDs from the get favorites call, get all quizzes from the backend, and then filter the quizzes by the favorite IDs.

Does this make sense? Let me know if you have any questions.